File: app/src/main/java/com/example/videoplayer/viewpager2/ScreenSlidePagerAdapter.java
package com.example.videoplayer.viewpager2;

import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentActivity;
import androidx.viewpager2.adapter.FragmentStateAdapter;

import com.example.videoplayer.VideoFragment;

public class ScreenSlidePagerAdapter extends FragmentStateAdapter {
    public ScreenSlidePagerAdapter(FragmentActivity fa) {
        super(fa);
    }

    @Override
    public Fragment createFragment(int position) {
        return new VideoFragment();
    }

    @Override
    public int getItemCount() {
        return 3;
    }
}

File: app/src/main/java/com/example/videoplayer/viewpager2/ScreenSlidePagerAdapter.kt
package com.example.videoplayer.viewpager2

import android.os.Bundle
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentActivity
import androidx.viewpager2.adapter.FragmentStateAdapter
import com.example.videoplayer.VideoFragment

class ScreenSlidePagerAdapter(
    fa: FragmentActivity?,
    var generateList: List<VideoItem>
) : FragmentStateAdapter(fa!!) {
    override fun createFragment(position: Int): Fragment {

        val bundle = Bundle()
        bundle.putParcelable(VideoFragment.PLAY_BEAN, generateList[position])
        bundle.putString(VideoFragment.SOURCE, "source")
        bundle.putInt(VideoFragment.POSITION, position)

        return VideoFragment.newInstance(bundle)
    }

    override fun getItemCount(): Int {
        return generateList.size
    }
}